zebra-state/src/service.rs

```diff
@@ -270,6 +270,95 @@ impl StateService {
             state: IterState::NonFinalized(hash),
         }
     }
+
+    /// Return an iterator over the finalized blocks of the chain
+    /// starting (and descending) from the block identified by `height`.
+    ///
+    /// The block identified by `height` is included in the chain of blocks yielded
+    /// by the iterator.
+    pub fn chain_finalized(&self, height: block::Height) -> Iter<'_> {
+        Iter {
+            service: self,
+            state: IterState::Finalized(height),
+        }
+    }
+
+    /// Return a list of blocks ahead given a list of blocks.
+    ///
+    /// 1. Make sure we have the list of blocks provided in our chain.
+    /// 2. Get a max of 500 hashes ahead of the provided tip.
+    pub fn find_chain_hashes(
+        &self,
+        block_locator: Vec<block::Hash>,
```

Comment:
Let's use the same names as `FindBlocks`, to avoid confusion:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/protocol/internal/request.rs#L100

```suggestion
        known_blocks: Vec<block::Hash>,
```